src/main/java/com/launchdarkly/client/LDClient.java

```diff
@@ -33,46 +35,80 @@
    * Creates a new client instance that connects to LaunchDarkly with the default configuration. In most
    * cases, you should use this constructor.
    *
-   * @param apiKey the API key for your account
+   * @param apiKey        the API key for your account
+   * @param waitForMillis when set to greater than zero allows callers to block until the client
+   *                      has connected to LaunchDarkly and is properly initialized
    */
-  public LDClient(String apiKey) {
-    this(apiKey, LDConfig.DEFAULT);
+  public LDClient(String apiKey, Long waitForMillis) {
+    this(apiKey, LDConfig.DEFAULT, waitForMillis);
   }
 
   /**
    * Creates a new client to connect to LaunchDarkly with a custom configuration. This constructor
    * can be used to configure advanced client features, such as customizing the LaunchDarkly base URL.
    *
-   * @param apiKey the API key for your account
-   * @param config a client configuration object
+   * @param apiKey        the API key for your account
+   * @param config        a client configuration object
+   * @param waitForMillis when set to greater than zero allows callers to block until the client
+   *                      has connected to LaunchDarkly and is properly initialized
    */
-  public LDClient(String apiKey, LDConfig config) {
+  public LDClient(String apiKey, LDConfig config, Long waitForMillis) {
     this.config = config;
     this.requestor = createFeatureRequestor(apiKey, config);
     this.eventProcessor = createEventProcessor(apiKey, config);
 
+    if (config.offline || config.useLdd) {
+      logger.info("Starting LaunchDarkly client in offline mode");
+      setOffline();
```

Comment:
One of the changes in the go-client is that this bit of state is now immutable. If that's the case then we should eliminate `setOffline`  / `setOnline` and we no longer need a thread-safe `volatile` boolean for `offline`-- it should just be a final.